lte/gateway/c/core/oai/tasks/sgw/sgw_handlers.cpp

```diff
@@ -2241,67 +2462,75 @@ void sgw_send_release_access_bearer_response(
+          !(eps_bearer_ctxt.ue_ip_paa().ipv6_addr().empty())) {
+        is_ue_ipv6_pres = true;
       }
-      OAILOG_DEBUG_UE(
-          module, imsi64,
-          "Deleting tunnel for bearer_id %u ue addr %x enb_ip %x "
-          "s_gw_teid_S1u_S12_S4_up %x, enb_teid_S1u %x pgw_up_ip %x "
-          "pgw_up_teid %x "
-          "s_gw_ip_address_S5_S8_up %x"
-          "s_gw_teid_S5_S8_up %x \n ",
-          eps_bearer_ctxt->eps_bearer_id, ue_ipv4.s_addr, enb.s_addr,
-          eps_bearer_ctxt->s_gw_teid_S1u_S12_S4_up,
-          eps_bearer_ctxt->enb_teid_S1u, pgw.s_addr,
-          eps_bearer_ctxt->p_gw_teid_S5_S8_up,
-          eps_bearer_ctxt->s_gw_ip_address_S5_S8_up.address.ipv4_address.s_addr,
-          eps_bearer_ctxt->s_gw_teid_S5_S8_up);
+
+      struct in_addr pgw = {.s_addr = 0};
+      struct in6_addr pgw_ipv6 = {};
+      convert_proto_ip_to_standard_ip_fmt(
+          eps_bearer_ctxt.mutable_pgw_address_up(), &pgw, &pgw_ipv6, true);
+
+      OAILOG_DEBUG_UE(module, imsi64,
+                      "Deleting tunnel for bearer_id %u ue addr %x enb_ip %x "
+                      "sgw_teid_s1u_s12_s4_up %x, enb_teid_S1u %x pgw_up_ip %x "
+                      "pgw_up_teid %x "
+                      "s_gw_ip_address_S5_S8_up %s"
+                      "s_gw_teid_S5_S8_up %x \n ",
+                      eps_bearer_ctxt.eps_bearer_id(), ue_ipv4.s_addr,
+                      enb.s_addr, eps_bearer_ctxt.sgw_teid_s1u_s12_s4_up(),
+                      eps_bearer_ctxt.enb_teid_s1u(), pgw.s_addr,
+                      eps_bearer_ctxt.pgw_teid_s5_s8_up(),
+                      eps_bearer_ctxt.sgw_ip_address_s5_s8_up().c_str(),
+                      eps_bearer_ctxt.sgw_teid_s5_s8_up());
 #if !MME_UNIT_TEST  // skip tunnel deletion for unit tests
+      int rv = RETURNok;
       if (module == LOG_SPGW_APP) {
-        rv = gtp_tunnel_ops->del_tunnel(
-            enb, enb_ipv6, eps_bearer_ctxt->paa.ipv4_address, ue_ipv6,
-            eps_bearer_ctxt->s_gw_teid_S1u_S12_S4_up,
-            eps_bearer_ctxt->enb_teid_S1u, NULL);
+        if (is_ue_ipv6_pres) {
+          rv = gtp_tunnel_ops->del_tunnel(
+              enb, &enb_ipv6, ue_ipv4, &ue_ipv6,
+              eps_bearer_ctxt.sgw_teid_s1u_s12_s4_up(),
+              eps_bearer_ctxt.enb_teid_s1u(), nullptr);
+        } else {
+          rv = gtp_tunnel_ops->del_tunnel(
+              enb, &enb_ipv6, ue_ipv4, nullptr,
+              eps_bearer_ctxt.sgw_teid_s1u_s12_s4_up(),
+              eps_bearer_ctxt.enb_teid_s1u(), nullptr);
+        }
       } else if (module == LOG_SGW_S8) {
-        rv = gtpv1u_del_s8_tunnel(enb, enb_ipv6, pgw, pgw_ipv6,
-                                  eps_bearer_ctxt->paa.ipv4_address, ue_ipv6,
-                                  eps_bearer_ctxt->s_gw_teid_S1u_S12_S4_up,
-                                  eps_bearer_ctxt->s_gw_teid_S5_S8_up);
+        rv = gtpv1u_del_s8_tunnel(enb, &enb_ipv6, pgw, &pgw_ipv6, ue_ipv4,
```

Comment:
Similar tothe check in line 2518, shouldn`t you check is_ue_ipv6_pres before invoking gtpv1u_del_s8_tunnel?

Reply:
Shall be taken care while porting structures for sgw_s8 task